src/babel-target.ts

```diff
@@ -245,14 +243,14 @@ export class BabelTargetFactory {
     return {
       presets: [
         ['@babel/preset-env', mergedPresetOptions],
+        ...this.presets,
```

Comment:
Have you checked this when `preset` is not set? Node generally doesn't like destructuring `undefined`

Reply:
Actually, looks like you set it to an empty array in the main plugin - in that case, it shouldn't be an optional param.